cmd/kubelet/app/server.go

```diff
@@ -339,6 +227,88 @@ func validateConfig(s *options.KubeletServer) error {
 	return nil
 }
 
+// if the node config controller was used, it will be returned as the first return value
+func initNodeConfig(s *options.KubeletServer, client clientset.Interface, nodeName string, standaloneMode bool) (*nodeconfig.NodeConfigController, error) {
```

Comment:
Where's the potential second global mutation (ofc updating the feature gates is one)?
Yeah no problem moving those state changes back to run().